File: windows/KeyMap.cs
using System.Collections.Generic;

namespace KeyboardDrop;

public static class KeyMap
{
    public static readonly Dictionary<string, int> NameToCode = new()
    {
        // Letters
        {"a", 0x41}, {"b", 0x42}, {"c", 0x43}, {"d", 0x44}, {"e", 0x45},
        {"f", 0x46}, {"g", 0x47}, {"h", 0x48}, {"i", 0x49}, {"j", 0x4A},
        {"k", 0x4B}, {"l", 0x4C}, {"m", 0x4D}, {"n", 0x4E}, {"o", 0x4F},
        {"p", 0x50}, {"q", 0x51}, {"r", 0x52}, {"s", 0x53}, {"t", 0x54},
        {"u", 0x55}, {"v", 0x56}, {"w", 0x57}, {"x", 0x58}, {"y", 0x59}, {"z", 0x5A},
        // Numbers
        {"0", 0x30}, {"1", 0x31}, {"2", 0x32}, {"3", 0x33}, {"4", 0x34},
        {"5", 0x35}, {"6", 0x36}, {"7", 0x37}, {"8", 0x38}, {"9", 0x39},
        // Symbols
        {"minus", 0xBD}, {"equal", 0xBB},
        {"left_bracket", 0xDB}, {"right_bracket", 0xDD},
        {"backslash", 0xDC}, {"semicolon", 0xBA}, {"quote", 0xDE},
        {"grave", 0xC0}, {"comma", 0xBC}, {"period", 0xBE}, {"slash", 0xBF},
        // Whitespace & control
        {"tab", 0x09}, {"space", 0x20}, {"return", 0x0D},
        {"escape", 0x1B}, {"delete", 0x08}, {"forward_delete", 0x2E},
        // Modifiers
        {"caps_lock", 0x14},
        {"left_shift", 0xA0}, {"right_shift", 0xA1},
        {"left_control", 0xA2}, {"right_control", 0xA3},
        {"left_option", 0xA4}, {"right_option", 0xA5},
        {"left_command", 0x5B}, {"right_command", 0x5C},
        {"left_win", 0x5B}, {"right_win", 0x5C},
        // Function keys
        {"f1", 0x70}, {"f2", 0x71}, {"f3", 0x72}, {"f4", 0x73},
        {"f5", 0x74}, {"f6", 0x75}, {"f7", 0x76}, {"f8", 0x77},
        {"f9", 0x78}, {"f10", 0x79}, {"f11", 0x7A}, {"f12", 0x7B},
        // Arrow keys
        {"up_arrow", 0x26}, {"down_arrow", 0x28},
        {"left_arrow", 0x25}, {"right_arrow", 0x27},
        // Navigation
        {"home", 0x24}, {"end", 0x23},
        {"page_up", 0x21}, {"page_down", 0x22},
    };

    public static readonly Dictionary<int, string> CodeToName = new();

    static KeyMap()
    {
        foreach (var kvp in NameToCode)
        {
            if (!CodeToName.ContainsKey(kvp.Value))
                CodeToName[kvp.Value] = kvp.Key;
        }
    }

    public static int? CodeFor(string name)
    {
        return NameToCode.TryGetValue(name.ToLowerInvariant(), out var code) ? code : null;
    }

    public static string NameFor(int code)
    {
        return CodeToName.TryGetValue(code, out var name) ? name : $"0x{code:X2}";
    }
}

File: macos/KeyMap.swift
import Foundation
import CoreGraphics

enum KeyMap {
    static let nameToCode: [String: CGKeyCode] = [
        // Letters
        "a": 0x00, "s": 0x01, "d": 0x02, "f": 0x03, "h": 0x04,
        "g": 0x05, "z": 0x06, "x": 0x07, "c": 0x08, "v": 0x09,
        "b": 0x0B, "q": 0x0C, "w": 0x0D, "e": 0x0E, "r": 0x0F,
        "y": 0x10, "t": 0x11, "o": 0x1F, "u": 0x20, "i": 0x22,
        "p": 0x23, "l": 0x25, "j": 0x26, "k": 0x28, "n": 0x2D,
        "m": 0x2E,
        // Numbers
        "0": 0x1D, "1": 0x12, "2": 0x13, "3": 0x14, "4": 0x15,
        "5": 0x17, "6": 0x16, "7": 0x1A, "8": 0x1C, "9": 0x19,
        // Symbols
        "minus": 0x1B, "equal": 0x18,
        "left_bracket": 0x21, "right_bracket": 0x1E,
        "backslash": 0x2A, "semicolon": 0x29, "quote": 0x27,
        "grave": 0x32, "comma": 0x2B, "period": 0x2F, "slash": 0x2C,
        // Whitespace & control
        "tab": 0x30, "space": 0x31, "return": 0x24,
        "escape": 0x35, "delete": 0x33, "forward_delete": 0x75,
        // Modifiers
        "caps_lock": 0x39,
        "left_shift": 0x38, "right_shift": 0x3C,
        "left_control": 0x3B, "right_control": 0x3E,
        "left_option": 0x3A, "right_option": 0x3D,
        "left_command": 0x37, "right_command": 0x36,
        "fn": 0x3F,
        // Function keys
        "f1": 0x7A, "f2": 0x78, "f3": 0x63, "f4": 0x76,
        "f5": 0x60, "f6": 0x61, "f7": 0x62, "f8": 0x64,
        "f9": 0x65, "f10": 0x6D, "f11": 0x67, "f12": 0x6F,
        // Arrow keys
        "up_arrow": 0x7E, "down_arrow": 0x7D,
        "left_arrow": 0x7B, "right_arrow": 0x7C,
        // Navigation
        "home": 0x73, "end": 0x77,
        "page_up": 0x74, "page_down": 0x79,
    ]

    static let codeToName: [CGKeyCode: String] = {
        var dict: [CGKeyCode: String] = [:]
        for (name, code) in nameToCode {
            dict[code] = name
        }
        return dict
    }()

    static func code(for name: String) -> CGKeyCode? {
        nameToCode[name.lowercased()]
    }

    static func name(for code: CGKeyCode) -> String? {
        codeToName[code]
    }
}
